pyrsia_cli/src/cli/handlers.rs

```
@@ -93,14 +98,17 @@ pub fn config_edit(host_name: Option<String>, port: Option<String>, diskspace: O
                         println!("Error Saving Node Configuration:       {}", error);
                     }
                 }
+                Ok(())
             } else {
                 errors.into_iter().for_each(|x| println!("{}", x));
-            }
+                Err(anyhow!("Invalid pyrsia config"))
+            };
         }
         Err(error) => {
             println!("Error Saving Node Configuration:       {}", error);
+            Err(anyhow!("Error Saving Node Configuration:       {}", error))
```

Instead of both printing and returning the error, we can just return the error and print it where the `config_edit` method is invoked?

---

changed as suggested.